### PR TITLE
Improved formatting of special values

### PR DESCRIPTION
**Pull request by tomgrubbmath**
This deals with Issue #2773. The 'display_complex()' function occasionally returned a string in scientific notation, which was not displaying correctly. I dealt with this by 

1.) Writing a function 'scientific_notation_helper' in Lfunctionutilities.py to look for the letter 'e' and replace it with '\\times 10^{}.'
2.) Importing this function into Lfunction_base.py so that it may be called on svt_crit and svt_edge
3.) Modifying templates/Lfunction.html to make sure dollar signs were used when needed. 

The 'before and after' can be seen best on this example: https://www.lmfdb.org/L/Character/Dirichlet/971/870/

### CODE REVIEW

File: lmfdb/lfunctions/Lfunctionutilities.py [hunk 1]
```diff
@@ -740,6 +740,34 @@ def specialValueTriple(L, s, sLatex_analytic, sLatex_arithmetic):
...
+            l_beg = lval_string[:efind1]
+            l_mid = lval_string[efind1+1:efind2]
+            l_end = lval_string[efind2+1:]
+            new_real = l_beg+'\\times 10^{'+l_mid[:l_mid.find(' ')]+'}'
+            new_im = l_mid[l_mid.find(' '):] + '\\times 10^{' + l_end[:l_end.find('i')]+'}i'
+            new_l_string = new_real+new_im
+        else:
+            l_beg = lval_string[:efind1]
+            l_end = lval_string[efind1+1:]
+            new_real = l_beg + '\\times 10^{' + l_end[:l_end.find(' ')]+'}'
```
Comment (roed314) [inline lmfdb/lfunctions/Lfunctionutilities.py:763]: Is there always a space at the end?  What if there's something useful after the space?

I worry that this code is too sensitive to the exact format of the input.  What about something like the following?
```python
lval_string = re.sub(r"[Ee](-?\d+)", r" \times 10^{\1}", lval_string)
```

Comment (davidfarmer) [reply lmfdb/lfunctions/Lfunctionutilities.py:763]: The regular expression substitution looks good to me, although it is always
necessary to check these things on actual examples.

For the math mode delimiters, I suggest changing them to \( and \),
which is more commonly used in the LMFDB (and perhaps we should
consider only using those and adding that to the coding standards).

Comment (tomgrubbmath) [reply lmfdb/lfunctions/Lfunctionutilities.py:763]: Thanks for the comments. @roed314 I had not worked with Python's re library before so I built it specific to the output of display_complex; I think yours looks much cleaner and more robust; I will do some tests as David suggests and then incorporate. @davidfarmer I will switch to the \( and \) standard. 

Sorry for the bad PR.

Comment (roed314) [reply lmfdb/lfunctions/Lfunctionutilities.py:763]: It's not a bad PR!  We're just trying to make it better. 😄

**Comment by tomgrubbmath** [comment]
PR updated to incorporate improvements from @davidfarmer and @roed314 ; thanks for your help! I learned something new today.

Changes:

Used David Roe's re.sub() idea to simplify the parsing/replacement function. It needed an extra backslash in the replaced string to properly display the \times character. 
For my own education, it works like this:
re.sub(r"[Ee](-?\d+)",r"\\times10^{\1}", lval_string): r prefix is for string literals. Brackets [Ee] matches an instance of "E" or "e;" parens () form a group later backreferenced by \1 in the replacement ; -? matches a negative sign if it is there, and \d+ matches the full string of digits afterwards. 

The other change replaces dollar signs with parenthetical delimiters following David Farmer's advice.

**Pull request merged**